src/read.c

```diff
@@ -1481,6 +1481,23 @@ static avifBool avifParseItemPropertyAssociation(avifMeta * meta, const uint8_t
                 }
             }
             if (supportedType) {
+                if (!essential) {
+                    // Verify that it is legal for this property to not be flagged as essential. Any
+                    // types in this list are *required* in the spec to be flagged as essential when
+                    // associated with an item.
+                    static const char * essentialTypes[] = {
+                        "av1C" // AVIF: Section 2.2.1: "This property shall be marked as essential."
```

Comment:
```suggestion
                        "a1op" // AVIF: Section 2.3.2.1: "If associated, it shall be marked as essential."
```
Not sure if `a1op` needs to be handled differently here.

Also, should there be an opposite check for `a1lx` since the spec says "[If associated, it shall not be marked as essential](https://aomediacodec.github.io/av1-avif/#layered-image-indexing-property-description)"

Reply:
Hmm. I'm thinking about removing `lsel` from the list for now and just adding boxes to this list as I add support for new boxes. `lsel` isn't written or read at all right now, so it seems odd to preemptively block a read on it.